zerver/views/auth.py

```diff
@@ -241,6 +243,7 @@ def maybe_send_to_registration(
             streams_to_subscribe = list(multiuse_obj.streams.all())
             prereg_user.streams.set(streams_to_subscribe)
             prereg_user.invited_as = invited_as
+            prereg_user.multiuse_invite = multiuse_obj
```

Comment:
Hmm, shouldn't this instead be passed to `create_preregistration_user`?

Reply:
@timabbott Hmm, good point - it wouldn't be enough right now, but maybe that's because the code above does weird things. First it is attempted to just grab a pre-existing PreregistrationUser:
```
            prereg_user = filter_to_valid_prereg_users(
                PreregistrationUser.objects.filter(email__iexact=email, realm=realm)
            ).latest("invited_at")
```
and `create_preregistration_user` is only called if the above fails:
```
        except PreregistrationUser.DoesNotExist:
            prereg_user = create_preregistration_user(
```

So I think we want `.multiuse_invite` set in either case? I'm not sure what's the reason that a pre-existing PreregistrationUser is attempted to be used though, rather than always creating a new one

Reply:
Ah, okay so this behavior of trying to grab a pre-existing prereg_user used to be only a `settings.ONLY_SSO` thing and I changed it to be done in all cases in 7ee54810a1eba115f376fb2d7d530b60c7ebd9b1 The original change implementing this is 54d72d9a55a5f1de5bb56c052d31ddddc2520b95 though I don't understand why it was necessary in order to `Fix Invitations on ONLY_SSO deploys` :thinking: 

I would vote to try to get rid of this and always `create_preregistration_user` - unless we see a reason to keep it as it is

Reply:
I think the reason is basically: If you setup your organization with SSO and then wanted to invite people to join specific streams using invitation links, then it's pretty annoying if someone just signs up ignoring the invitation link via SSO.

It'd be better to prompt, but I think we can leave this as-is until we do so.

Reply:
Merged with a change to set this in the `create_preregistartion_user` call when we're doing this -- will be one less thing to check if we do refactor to remove this feature.